File: ConsoleTests/ListChoiceReaderTest.vb
Public Class ListChoiceReaderTest

    <InteractiveRunnable(True)>
    Public Shared Sub Start()
        Dim Rand = New Random()
        Dim L = Utilities.Collections.Range(45).Select(Of Integer?)(Function(I) Rand.Next())
        Dim ChoiceReader = New ConsoleListChoiceReader(Of Integer?)(L)

        Do
            Dim I = ChoiceReader.ReadChoice()
            If Not I.HasValue Then
                Exit Do
            End If
            Console.WriteLine($"{I} was chosen!!")
        Loop
    End Sub

End Class

File: ConsoleTests/ListChoiceReaderTest.cs
using System;
using System.Linq;

using Ks.Common;

namespace Ks.ConsoleTests
{
    public static class ListChoiceReaderTest
    {
        [InteractiveRunnable(true)]
        public static void Start()
        {
            var Rand = new Random();
            var L = Utilities.Collections.Range(45).Select(I => (int?) Rand.Next());
            var ChoiceReader = new ConsoleListChoiceReader<int?>(L);

            while (true)
            {
                var I = ChoiceReader.ReadChoice();
                if (!I.HasValue)
                {
                    break;
                }

                Console.WriteLine($"{I} was chosen!!");
            }
        }
    }
}
